browser/ui/tabs/shared_pinned_tab_service.cc

```diff
@@ -357,13 +326,19 @@ void SharedPinnedTabService::OnBrowserClosing(Browser* browser) {
       }
     }
   } else {
+    CHECK(!profile_will_be_destroyed_);
+
     // Try caching shared contents from the closing browser.
     auto* tab_strip_model = browser->tab_strip_model();
```

Comment:
this entire class is problematic and it should not be keeping references to `Browser` like this. Need to open an issue to fix this.

Reply:
cc: @simonhong

Reply:
Filed https://github.com/brave/brave-browser/issues/42447